DASH-IF-IOP.bs.md

```diff
@@ -10,8 +10,500 @@ Live discussion in #document-authoring on Slack.
+
+1. The MPD consists of consecutive [=periods=] which map data onto the MPD timeline.
+1. Each period contains of one or more [=representations=], each of which provides media samples inside a sequence of [=media segments=].
+1. Representations within a period are grouped in [=adaptation sets=], which associate related representations and decorate them with metadata.
+
+<figure>
+	<img src="Images/Timing/BasicMpdElements.png" />
+	<figcaption>The primary elements described by an MPD.</figcaption>
+</figure>
+
+The chapters below explore these relationships in detail.
+
+## Periods ## {#timing-period}
+
+An MPD SHALL define an ordered list of one or more <dfn title="period">periods</dfn>. A period is both a time span on the [=MPD timeline=] and a definition of the data to be presented during the period. Period timing is relative to the zero point of the [=MPD timeline=].
+
+<figure>
+	<img src="Images/Timing/PeriodsMakeTheMpd.png" />
+	<figcaption>An MPD is a collection of periods, starting from a zero point and with a total duration (which may be unlimited for live services).</figcaption>
+</figure>
+
+Common reasons for defining multiple periods are:
+
+* Assembling a presentation from multiple self-contained pieces of content.
+* Inserting ads in the middle of existing content and/or replacing spans of existing content with ads.
+* Adding/removing certain representations as the nature of the content changes (e.g. a new title starts with a different set of offered languages).
+* Updating period-scoped metadata (e.g. codec configuration or DRM signaling).
+
+Perioods are self-contained - a service SHALL NOT require a client to know the contents of another period in order to correctly present a period. Knowledge of the contents of different periods MAY be used by a client to achieve seamless period transitions, especially when working with [[#timing-connectivity|period-connected representations]].
+
+All periods SHALL be consecutive and non-overlapping. A period MAY have a duration of zero.
+
+Note: A period with a duration of zero might, for example, be the result of ad-insertion logic deciding not to insert any ad.
+
+<div class="example">
+The below MPD example consists of two 20-second periods. The duration of the first period is calculated using the start point of the second period.
+
+<xmp highlight="xml">
+<MPD type="static" mediaPresentationDuration="PT40S">
+	<Period>
+		...
+	</Period>
+	<Period start="PT20S" duration="PT20S">
+		...
+	</Period>
+</MPD>
+</xmp>
+</div>
+
+In a static MPD, the first period SHALL start at the zero point of the [=MPD timeline=]. In a dynamic MPD, the first period SHALL start at or after the zero point of the [=MPD timeline=].
+
+In a static MPD, the last period SHALL have a `Period@duration`. In a dynamic MPD, the last period MAY lack a `Period@duration`, in which case it SHALL be considered to have an unlimited duration.
```

Comment:
You can connect `Period@duration` to extensibility of the period though, which is nicely analogous to connecting `mediaPresentationDuration` to the extensibility of the presentation. The way this is worded currently doesn't clearly define when `Period@duration` should/shouldn't be set.

Reply:
Oh, I see what you mean now. Yeah, makes sense. I will do so.

Reply:
There seems more here to consider.

You can have a fixed-duration period that is not yet completely filled with segments because the period extends beyond t=now (segments after the live edge do not yet need to be referenced). Requiring `Period@duration` to mean "I will no longer add segment references" would conflict with that.

I think we cannot assume that a duration means "no more segment references". It just means "the period will not extend beyond this point" but the period may still be in the process of being filled with segments.

I think we can assume that MPD@mediaPresentationDuration means "no more periods are coming" though.

What does this mean for the player? It can only use durations to see where periods are. It cannot assume that just because there is a duration, all the segment references are in place. There might still be new segment references added. Removal of minimumUpdatePeriod would be the only signal that it doesn't need to worry about updates anymore.

Do you see any inconsistency in this line of reasoning?

Reply:
I think that makes sense to me.

> the period will not extend beyond this point

This leaves the possibility open for the period to be shortened. This probably depends on the outcome of https://github.com/Dash-Industry-Forum/DASH-IF-IOP/issues/224#issuecomment-442670474. If that's not relaxed then I think it should just say that the period duration cannot change once it's set, and that it shall be set once the duration of the period is known.

Reply:
I will leave it on the "shortening is OK" interpretation right now and adjust later if needed.